### PR TITLE
[B] Table based test

### LINKED ISSUE #134: [B] Table based tests
Scenario: User specifies that stored procedure should create/delete a table

Detect conditions related to a Table

- Does table exist
- Does table the # of records change by x

### PR DESCRIPTION
**Pull request by insaan21**
# Description

Created table based test to check if table exists, and count the number of rows in a table

## Link to the issue you are closing

Fixes #134 

## Screenshot or other testing that you have completed

Screen Shot 2022-10-14 at 5 24 57 AM
Screen Shot 2022-10-14 at 5 25 16 AM
Screen Shot 2022-10-14 at 5 25 44 AM
Screen Shot 2022-10-14 at 5 25 49 AM
Screen Shot 2022-10-14 at 5 37 41 AM
Screen Shot 2022-10-14 at 5 37 51 AM

Your PR should include at least one test/screenshot in order to showcase that the changes don't break/fix what was needed. In the case of frontend, a screenshot should suffice. For backend, depending on the scenario, some temporary tests may need to be written to show that the function does what's needed.

# Checklist:

- [x] ESLint passes
- [x] Prettier passes
- [x] Included comments where necessary
- [x] Updated README if needed
- [x] Verified that code runs and generates no errors/warnings

### CODE REVIEW

File: src/main/ipc.ts [hunk 1]
```diff
@@ -20,6 +20,14 @@ ipcMain.handle('procedures:getProcedure', async (_event, ...args) => {
...
 });
 
+ipcMain.handle('procedures:checkTableExists', async (_event, ...args) => {
+  return new Procedures().checkTableExists(args[0]);
+});
+
+ipcMain.handle('procedures:numRecordsInTable', async (_event, ...args) => {
+  return new Procedures().numRecordsInTable(args[0]);
+});
+
```
Comment (adityavidyadharan) [inline src/main/ipc.ts:30]:
```suggestion
```

I don't think these functions need to go in the ipc renderer. The front end will just be passing the RuleType model to the back end and then based on the fields within, the backend will choose the correct function to call and return a response from (as you are doing in pass-fail)

File: src/main/preload.ts [hunk 2]
```diff
@@ -38,6 +38,10 @@ contextBridge.exposeInMainWorld('procedures', {
     fetchDatabases: () => ipcRenderer.invoke('procedures:listDatabases'),
     fetchContent: (procedure: string) =>
       ipcRenderer.invoke('procedures:getProcedure', procedure),
+    checkTableExists: (table: string) =>
+      ipcRenderer.invoke('procedures:checkTableExists', table),
+    numRecordsInTable: (table: string) =>
+      ipcRenderer.invoke('procedures:numRecordsInTable', table),
```
Comment (adityavidyadharan) [inline src/main/preload.ts:44]:
```suggestion
```
Same as above

File: src/renderer/preload.d.ts [hunk 3]
```diff
@@ -33,6 +33,8 @@ declare global {
         fetchProcedures(): Promise<Map<string, string[]>>;
         fetchDatabases(): Promise<string[]>;
         fetchContent(procedure: string): Promise<string>;
+        checkTableExists(table: string): Promise<boolean>;
+        numRecordsInTable(table: string): Promise<number>;
```
Comment (adityavidyadharan) [inline src/renderer/preload.d.ts:37]:
```suggestion
```

**Review by adityavidyadharan — CHANGES_REQUESTED** [review]
Methods look good but i dont think we need the IPC stuff for this

File: src/main/ipc.ts (hunk 1 above)
Comment (insaan21) [reply src/main/ipc.ts:30]: Yeah makes sense

**Review by adityavidyadharan — APPROVED** [review]

**Pull request merged**